### ISSUE TITLE
code editor: make it `invalid` when the value is invalid

### ISSUE THREAD

**Issue by Kiarokh** [Lundalogik/lime-elements#3764]
When the code inside the Code Editor is invalid (e.g. contains lint errors), the component renders a visual element `<div class="CodeMirror-gutter-elt"` inside `<div class="CodeMirror-gutter-wrapper"`. 

Now that the component has an `invalid` prop: https://github.com/Lundalogik/lime-elements/issues/3763 we can use this internal validation mechanisms to set the component to invalid, when its value is invalid. For example, when the value contains lint errors.

The `invalid` prop in `limel-code-editor` has proper accessibility support, and is better wired for indicating the invalid state.

<img width="919" height="191" alt="Image" src="https://github.com/user-attachments/assets/1dc348a2-220f-4e1c-a08d-d5c6254916c6" />

That alert of "Should be a valid JSON" doesn't need to be displayed like this anymore:

https://github.com/Lundalogik/lime-elements/blob/1d52e26e42ee7ce3d983086db1b26de36c2cdb48/src/components/form/widgets/code-editor.ts#L49

We need to revisit this form widget, and instead use `helperText`, 

https://github.com/Lundalogik/lime-elements/blob/main/src/components/form/widgets/code-editor.ts
